docs/src/client_configuration.md

```diff
@@ -42,7 +42,7 @@ Below is an example of the `LSP.sublime-settings` file with configurations for t
 | initializationOptions | options to send to the server at startup (rarely used) |
 | selector | This is _the_ connection between your files and language servers. It's a selector that is matched against the current view's base scope. If the selector matches with the base scope of the the file, the associated language server is started. For more information, see https://www.sublimetext.com/docs/3/selectors.html |
 | priority_selector | Used to prioritize a certain language server when choosing which one to query on views with multiple servers active. Certain LSP actions have to pick which server to query and this setting can be used to decide which one to pick based on the current scopes at the cursor location. For example when having both HTML and PHP servers running on a PHP file, this can be used to give priority to the HTML one in HTML blocks and to PHP one otherwise. That would be done by setting "feature_selector" to `text.html` for HTML server and `source.php` to PHP server. Note: when the "feature_selector" is missing, it will be the same as the "document_selector".
-| hide_non_project_diagnostics | Enable to ignore diagnostics for files that are not within the project (window) folders. If project has no folders then this option has no effect and diagnostics are shown for all files. |
+| diagnostics_mode | Set to `"workspace"` to ignore diagnostics for files that are not within the project (window) folders. If project has no folders then this option has no effect and diagnostics are shown for all files. If the server supports _pull diagnostics_ (`diagnosticProvider`), this setting controls whether diagnostics are requested only for open files (`"open_files"`), or for all files in the project folders (`"workspace"`). |
```

Comment:
It's not quite clear what's the default behavior here for both non-pull and pull diagnostics.

Reply:
Not sure if it's better now, or if you meant something else.
To be honest, I'm not 100% happy with the option names I used for this setting, because "open_files" could be a bit misleading, in the way that you might wrongly expect diagnostics for files in the workspace folders not being shown, if the file is not open. But I don't really have a better idea for the option name(s), and I still tend to think that having two different settings for this (one for "hide_non_project_diagnostics" and another one for "pull_diagnostics_mode" (controlling e.g. whether to clear the diagnostics if a file gets closed)) would be unnecessary, because the use cases for the different values of those settings probably would directly overlap.